cmd/manager/main.go

```diff
@@ -4,140 +4,146 @@ import (
+	flag.StringVar(&tlsCertPath, "tls-cert", "", "Path to use for certificate (requires tls-key)")
 	flag.Parse()
 
 	// Check if version flag was set
-	if *version {
-		fmt.Print(sourceCommit.String())
-		// Exit immediately
+	if version {
+		logrus.Infof(sourceCommit.String())
 		os.Exit(0)
 	}
 
 	// set TLS to serve metrics over a secure channel if cert is provided
 	// cert is provided by default by the marketplace-trusted-ca volume mounted as part of the marketplace-operator deployment
-	var useTLS bool
-	if *tlsCertPath != "" && *tlsKeyPath == "" || *tlsCertPath == "" && *tlsKeyPath != "" {
-		log.Warn("both --tls-key and --tls-crt must be provided for TLS to be enabled, falling back to non-https")
-	} else if *tlsCertPath == "" && *tlsKeyPath == "" {
-		log.Info("TLS keys not set, using non-https for metrics")
-	} else {
-		log.Info("TLS keys set, using https for metrics")
-		useTLS = true
-	}
-	err := metrics.ServePrometheus(useTLS, *tlsCertPath, *tlsKeyPath)
+	err := metrics.ServePrometheus(tlsCertPath, tlsKeyPath)
 	if err != nil {
-		log.Fatalf("failed to serve prometheus metrics: TLS enabled %d: %s", useTLS, err)
+		logrus.Errorf("failed to serve prometheus metrics: %s", err)
+		os.Exit(1)
 	}
 
 	namespace, err := k8sutil.GetWatchNamespace()
 	if err != nil {
-		log.Fatalf("failed to get watch namespace: %v", err)
+		logrus.Errorf("failed to get watch namespace: %v", err)
+		os.Exit(1)
 	}
 
 	// Get a config to talk to the apiserver
 	cfg, err := config.GetConfig()
 	if err != nil {
-		log.Fatal(err)
+		logrus.Error(err)
+		os.Exit(1)
 	}
 
 	// Set OpenShift config API availability
 	err = configv1.SetConfigAPIAvailability(cfg)
 	if err != nil {
-		log.Fatal(err)
+		logrus.Error(err)
+		os.Exit(1)
 	}
 
 	// Create a new Cmd to provide shared dependencies and start components
 	// Even though we are asking to watch all namespaces, we only handle events
 	// from the operator's namespace. The reason for watching all namespaces is
 	// watch for CatalogSources in targetNamespaces being deleted and recreate
 	// them.
-	mgr, err := manager.New(cfg, manager.Options{Namespace: ""})
+	//
+	// Note(tflannag): Setting the `MetricsBindAddress` to `0` here disables the
+	// metrics listener from controller-runtime. Previously, this was disabled by
+	// default in <v0.2.0, but it's now enabled by default and the default port
+	// conflicts with the same port we bind for the health checks.
+	mgr, err := manager.New(cfg, manager.Options{
+		Namespace:              "",
+		HealthProbeBindAddress: ":8080",
+		MetricsBindAddress:     "0",
+	})
 	if err != nil {
-		log.Fatal(err)
+		logrus.Error(err)
+		os.Exit(1)
 	}
 
-	log.Print("Registering Components.")
-
+	logrus.Info("Registering Components.")
+	logrus.Info("setting up scheme")
 	// Setup Scheme for all defined resources
 	if err := apis.AddToScheme(mgr.GetScheme()); err != nil {
-		exit(err)
+		logrus.Error(err)
+		os.Exit(1)
```

Comment:
May as well replace all of these with https://pkg.go.dev/github.com/sirupsen/logrus?utm_source=godoc#Fatalf while you're changing this.